draft-ietf-httpbis-message-signatures.md

```diff
@@ -1048,6 +1051,49 @@ Signature: sig1=:P0wLUszWQjoi54udOtydf9IWTfNhy+r53jGFj9XZuP4uKwxyJo\
 
 The proxy's signature and the client's original signature can be verified independently for the same message, depending on the needs of the application.
 
+# Requesting Signatures
+
+While a signer is free to attach a signature to a request or response without prompting, it is often desirable for a potential verifier to signal that it expects a signature from a potential signer using the `Accept-Signature` field.
+
+The message to which the requested signature is applied is known as the "target message". When the `Accept-Signature` field is sent in an HTTP Request message, the field indicates that the client desires the server to sign the response using the identified parameters and the target message is the response to this request. The server can choose to also continue signing future responses to the same client in the same way. Responses that are the result of signature negotiation in this manner MUST either be marked as not cacheable or contain a `Vary` field, in order to prevent a proxy from returning a response with a signature intended for a different request.
```

Comment:
```suggestion
The message to which the requested signature is applied is known as the "target message". When the `Accept-Signature` field is sent in an HTTP Request message, the field indicates that the client desires the server to sign the response using the identified parameters and the target message is the response to this request. All responses from resources that support such signature negotiation SHOULD either be uncacheable or contain a `Vary` header field that lists `Accept-Signature`, in order to prevent a cache from returning a response with a signature intended for a different request.
```

It's a SHOULD in the core HTTP specs, so making it a MUST here seems a bit too high of a bar. I deleted 'the server can choose' because it implies that HTTP is a stateful protocol.

Reply:
Thanks, this is a good change and I'll add it. Should we make a similar change in the next paragraph, which currently has:

> The client can choose to also continue signing future requests to the same server in the same way.

Or is it more acceptable in this direction because a client has more of an "identity" in mind for a given server it's talking to?